### ISSUE TITLE
Train result on my own dataset. A Big gap between Train and Valid dataset

### ISSUE THREAD

**Issue by Erichen911** [lucidrains/vit-pytorch#32]
Hey guys.
First of all. This is a great job and thanks to the authors.
Then my question is... 
Recently I use this code on my own dataset. A simple binary-classification problem.
The performance on the training dataset is good, but not as well as the validation dataset.
The Loss curve is...
![image](https://user-images.githubusercontent.com/35665175/100040932-32820380-2e43-11eb-8ca5-44c08aaaa0d3.png)

My model is 
model = ViT(
    dim=128,
    image_size=224,
    patch_size=32,
    num_classes=2,
    depth=12,
    heads=8,
    mlp_dim=512,
    channels=3,
)

Training Dataset has 1200+ images, Validation Dataset has 300+ images.

Can someone give me some suggestions, how to solve this problem?

I think there are several possibilities. Maybe I need a pretrained model? Or I did the wrong way in the training of transformer model?

**Comment by lucidrains** [comment]
@Erichen911 1200 is not enough! Off by 3 orders of magnitude at least!

**Comment by lucidrains** [comment]
@Erichen911 I would recommend getting a huge amount of images, preferrably a million at least, and then doing self-supervised learning with BYOL, before training on your tiny training set

otherwise, just use Ross' pretrained model!

**Comment by henbucuoshanghai** [comment]
@Erichen911   can you  share your code?> tks

**Comment by khawar512** [comment]
Does anyone train Swin transformers with different image sizes? Like my image size is 112x112, it never works on this size

**Comment by abhigoku10** [comment]
@lucidrains @Erichen911  can you share the train.py which you guys r using for custom data or any reference ??
